IOPool/Output/src/RootOutputFile.cc

```diff
@@ -608,11 +612,40 @@ namespace edm {
     b->Fill();
   }
 
+  // For duplicate removal and to determine if fast cloning is possible, the input
+  // module by default reads the entire EventAuxiliary branch when it opens the
+  // input files.  If EventAuxiliary is written in the usual way, this results
+  // in many small reads scattered throughout the file, which can have very poor
+  // performance characteristics on some filesystems.  As a workaround, we save
+  // EventAuxiliary data and the ProcessHistoryID, writing the branch as we are
+  // closing the file, resulting in (hopefully) one basket at the end of the file
+  // that can be accessed with one read.
+  void RootOutputFile::writeEventAuxiliary() {
+    auto tree = eventTree_.tree();
+    auto bname = BranchTypeToAuxiliaryBranchName(InEvent).c_str();
```

Comment:
I'm not sure where this is declared - can you double-check that you're not getting a pointer to a temporary?